Source/Tabs/DataProviders/ShoppingList.lua

```diff
@@ -0,0 +1,129 @@
+local SHOPPING_LIST_TABLE_LAYOUT = {
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "minPrice" },
+    headerText = "Price",
+    cellTemplate = "AuctionatorPriceCellTemplate",
+    cellParameters = { "minPrice" },
+    width = 140
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "name" },
+    headerText = "Name",
+    cellTemplate = "AuctionatorItemKeyCellTemplate"
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerText = "Available",
+    headerParameters = { "totalQuantity" },
+    cellTemplate = "AuctionatorStringCellTemplate",
+    cellParameters = { "totalQuantity" },
+    width = 70
+  }
+}
+
+local ITEM_KEY_EVENT = "ITEM_KEY_ITEM_INFO_RECEIVED"
+
+ShoppingListDataProviderMixin = CreateFromMixins(DataProviderMixin)
+
+function ShoppingListDataProviderMixin:OnLoad()
+  Auctionator.Debug.Message("ShoppingListDataProviderMixin:OnLoad()")
+
+  self.pendingItemIds = {}
+
+  DataProviderMixin.OnLoad(self)
+
+  self:GetParent():Register( self, {
+    Auctionator.ShoppingLists.Events.ListSearchStarted,
+    Auctionator.ShoppingLists.Events.ListSearchEnded,
+    Auctionator.ShoppingLists.Events.ListSearchIncrementalUpdate
+  })
+end
+
+function ShoppingListDataProviderMixin:OnEvent(event, ...)
+  if event == ITEM_KEY_EVENT then
+    local itemId = ...
+
+    for _, entry in ipairs(self.results) do
+      if entry.itemKey.itemID == itemId then
+        self:FetchItemKey(entry)
+        return
+      end
+    end
+  end
+end
+
+function ShoppingListDataProviderMixin:EventUpdate(eventName, eventData)
+  Auctionator.Debug.Message(eventName, eventData)
+
+  local fetchItemKey = function(rowEntry)
+    self:FetchItemKey(rowEntry)
+  end
+
+  if eventName == Auctionator.ShoppingLists.Events.ListSearchStarted then
+    self:Reset()
+  elseif eventName == Auctionator.ShoppingLists.Events.ListSearchEnded then
+    self:AppendEntries(eventData, fetchItemKey)
+  elseif eventName == Auctionator.ShoppingLists.Events.ListSearchIncrementalUpdate then
+    self:AppendEntries(eventData, fetchItemKey)
+  end
+end
+
+function ShoppingListDataProviderMixin:FetchItemKey(rowEntry)
+  local itemKeyInfo = C_AuctionHouse.GetItemKeyInfo(rowEntry.itemKey)
+
+  if not itemKeyInfo then
+    self:RegisterEvent(ITEM_KEY_EVENT)
+
+    table.insert(self.pendingItemIds, rowEntry.itemKey.itemID)
+    rowEntry.itemName = ""
+
+    return
+  end
+
+  if #self.pendingItemIds == 0 then
+    self:UnregisterEvent(ITEM_KEY_EVENT)
+  end
+
+  local text = AuctionHouseUtil.GetItemDisplayTextFromItemKey(rowEntry.itemKey, itemKeyInfo, false)
+
+  rowEntry.itemName = text
+  rowEntry.name = Auctionator.Utilities.RemoveTextColor(text)
+  rowEntry.iconTexture = itemKeyInfo.iconFileID
+  rowEntry.noneAvailable = rowEntry.totalQuantity == 0
+
+  self.onUpdate(self.results)
```

Comment:
Can it be sorted at this point (using whatever the last column settings are)?

Reply:
This will only happen as the search is progressing, so there should be no sort settings yet. Do you think we should probably disable the column header clicking until search has completed?